### PR TITLE
Finish Project

### PR DESCRIPTION
**Pull request by ShaDy03**
I was add class library Models with folders Request + Response
I wish delete Models.Response, but i can't.
I was finished Add-GetById-GetAll-Update-Delete on Project Table, i was testing and that's worked!

### CODE REVIEW

File: IssueTracker/DataAccess/DbAccess/ISQLDataAccess.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 {
     public interface ISQLDataAccess
     {
+        Task<IEnumerable<T>> LoadData<T>(string storedProcedure, string connectionId = "Default");
```
Comment (iiusan) [inline IssueTracker/DataAccess/DbAccess/ISQLDataAccess.cs]: missing 'async' suffix

File: IssueTracker/DataAccess/Models/Issue.cs [hunk 2]
```diff
@@ -24,3 +24,5 @@ public class Issue
         public IssueType? IssueType { get; set; }
     }
 }
+
+//TODO: Object FK Priority & Status & Role & Project
```
Comment (iiusan) [inline IssueTracker/DataAccess/Models/Issue.cs]: no unresolved TODOs in PRs

File: IssueTracker/IssueTrackerAPI/Controllers/ProjectController.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            _project = project;
+        }
+
+        [HttpPost("add-project")]
+        public async Task AddProject(ProjectRequest entity) =>
+            await _project.AddAsync(entity);
+
+        [HttpGet("getAll-project")]
+        public async Task<List<Project>> GetAll() =>
```
Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/Controllers/ProjectController.cs]: change to IEnumerable

File: IssueTracker/Models/Response/ProjectInfo.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Models.Info
+{
+    public class ProjectInfo
+    {
+        public ProjectInfo()
+        {
+            Participants = new();
```
Comment (iiusan) [inline IssueTracker/Models/Response/ProjectInfo.cs:7]: why new()?

File: IssueTracker/Models/Response/ParticipantInfo.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Models.Info
+{
+    public class ParticipantInfo
```
Comment (iiusan) [inline IssueTracker/Models/Response/ParticipantInfo.cs:3]: empty class

File: IssueTracker/DataAccess/Data/ProjectData.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    public class ProjectData : IProjectData
+    {
+        private readonly ISQLDataAccess _db;
+        public ProjectData(ISQLDataAccess db)
+        {
+            _db= db;
+        }
+
+        public async Task AddAsync(ProjectRequest project) =>
+            await _db.SaveData("dbo.spProject_Insert", new { project.Title, project.Description, DateTime.Now });
```
Comment (iiusan) [inline IssueTracker/DataAccess/Data/ProjectData.cs]: use UtcNow

File: IssueTracker/DataAccess/Data/IData/IProjectData.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+using Models.Request;
+
+namespace DataAccess.Data.IData
+{
+    public interface IProjectData
+    {
+        Task AddAsync(ProjectRequest project);
+        Task<IEnumerable<Project>> GetAllAsync();
+        Task<Project?> GetByIdAsync(int id);
+        Task UpdateAsync(ProjectRequest project);
```
Comment (iiusan) [inline IssueTracker/DataAccess/Data/IData/IProjectData.cs]: no mixing between api & db dtos

**Review by iiusan — CHANGES_REQUESTED** [review]

File: IssueTracker/DataAccess/DbAccess/ISQLDataAccess.cs (hunk 1 above)
Comment (ShaDy03) [reply IssueTracker/DataAccess/DbAccess/ISQLDataAccess.cs]: can i put async in interface?

File: IssueTracker/Models/Response/ProjectInfo.cs (hunk 4 above)
Comment (ShaDy03) [reply IssueTracker/Models/Response/ProjectInfo.cs:7]: i want initializer the list

**Review by iiusan — APPROVED** [review]

**Pull request merged**